internal/controller/azurevalidator_controller.go

```diff
@@ -65,77 +61,51 @@ func (r *AzureValidatorReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	// With the active validator, we can then look for its validation
-	// result. So we try to get the validation result out of k8s. It might
-	// not exist yet.
-	//
-	// If we find that it *does* exist, we handle it as an existing validation
-	// result, which means to just do some logging with the data in it. It
-	// doesn't involve doing more reading from k8s, writing to k8s, etc.
-	//
-	// If we find that it *does not* exist, we handle it as a new validation
-	// result, which means to create a new validation result object, update
-	// its status, save it in k8s. Its status is a "state" and "conditions". We
-	// use "InProgress" as the state because this is a brand new validation
-	// result object and we haven't done anything meaningful yet. We use an
-	// an empty set of conditions for the same reason.
-	vr := &v8or.ValidationResult{}
+	// Get the active validator's validation result
+	vr := &vapi.ValidationResult{}
 	nn := ktypes.NamespacedName{
 		Name:      fmt.Sprintf("validator-plugin-azure-%s", validator.Name),
 		Namespace: req.Namespace,
 	}
 	if err := r.Get(ctx, nn, vr); err == nil {
-		res, err := v8ores.HandleExistingValidationResult(nn, vr, r.Log)
+		res, err := vres.HandleExistingValidationResult(nn, vr, r.Log)
```

Comment:
I simplified `HandleExistingValidationResult` and `HandleNewValidationResult` in the core validator repo because they were overly complex. You may want to get the latest version and update this section accordingly. See the AWS plugin for reference.

Reply:
Addressed by updating to latest validator dependency and making the changes in a commit to be compatible.